src/xml2_url.cpp

```diff
@@ -1,156 +1,133 @@
+#include <cpp11.hpp>
+
 #define R_NO_REMAP
 #include <Rinternals.h>
 #undef R_NO_REMAP
 
 #include <libxml/uri.h>
 #include "xml2_utils.h"
 
-// [[export]]
-extern "C" SEXP url_absolute_(SEXP x_sxp, SEXP base_sxp) {
-  R_xlen_t n = Rf_xlength(x_sxp);
-  SEXP out = PROTECT(Rf_allocVector(STRSXP, n));
-
-  if (Rf_xlength(base_sxp) > 1) {
-    Rf_error("Base URL must be length 1");
+const xmlChar* to_xml_chr(cpp11::strings x, const char* arg) {
+  if (x.size() > 1) {
```

Comment:
```suggestion
  if (x.size() != 1) {
```

Thinking about empty vectors too